File: 99-Continue/156/InOrderTraversal.py
from collections import deque 

class Node:
    def __init__(self,data:int):
        self.left = None
        self.right = None
        self.data = data

class BinaryTree:
    def __init__(self):
        self.root = None

    def buildTree(self, n:int)->Node:
        values = input().split()
        if not values:
            return None
        self.root = Node(values[0])
        queue = deque([self.root])
        i = 1
        while(queue and i<len(values)):
            current = queue.popleft()
            if(i<len(values)):
                if values[i]!="null":
                    current.left = Node(values[i])
                    queue.append(current.left)
                i+=1

            if(i<len(values)):
                if values[i]!="null":
                    current.right= Node(values[i])
                    queue.append(current.right)
                i+=1
        return self.root

    def inOrderTraversal(self,root:Node):
        if root:
            self.inOrderTraversal(root.left)
            print(root.data,end=" ")
            self.inOrderTraversal(root.right)

def main():
    n = int(input())
    tree = BinaryTree()
    tree.buildTree(n)
    tree.inOrderTraversal(tree.root)

if __name__ =="__main__":
    main()

File: 99-Continue/156/InOrderTraversal.java
import java.util.*;

public class InOrderTraversal {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int n = sc.nextInt();
        sc.nextLine();
        BinaryTree tree = new BinaryTree();
        tree.buildTree(n, sc);
        tree.inOrderTraversal(tree.root);
        sc.close();
    }
}

class BinaryTree{
    Node root;
    BinaryTree(){
        this.root = null;
    }
    
    Node buildTree(int n, Scanner sc) {
        String[] strarr = sc.nextLine().split(" ");

        if (strarr.length == 0)
            return null;

        root = new Node(Integer.parseInt(strarr[0]));

        Queue<Node> queue = new LinkedList<>();
        queue.offer(root);

        int i = 1;

        while (!queue.isEmpty() && i < strarr.length) {

            Node current = queue.poll();

            if (i < strarr.length) {
                if (!strarr[i].equals("null")) {
                    current.left = new Node(Integer.parseInt(strarr[i]));
                    queue.offer(current.left);
                }
                i++;
            }

            if (i < strarr.length) {
                if (!strarr[i].equals("null")) {
                    current.right = new Node(Integer.parseInt(strarr[i]));
                    queue.offer(current.right);
                }
                i++;
            }
        }

        return root;
    }
    
    void inOrderTraversal(Node root){
        if(root!=null){
            inOrderTraversal(root.left);
            System.out.print(root.val+" ");
            inOrderTraversal(root.right);
        }
    }
}

class Node{
    Node left,right;
    int val;
    Node(int val){
        this.val = val;
    }
}